### ISSUE TITLE
query_range only return 5 mins data

### ISSUE THREAD

**Issue by yrka5180** [jacksontj/promxy#599]
Problem:
I set up a promxy that covers a victoriametrics-cluster.
The problem is that when I query through Promxy only returns 5 mins data comparing with vm-select.

Config:
```
promxy:
  server_groups:
    - static_configs:
        - targets:
          - vm-select:9090
      anti_affinity: 15s
      timeout: 10s
      remote_read_path: api/v1/read
      path_prefix: /prometheus
      query_params:
        nocache: 1
        step: 15s
      scheme: http
      http_client:
        dial_timeout: 1s
        tls_config:
          insecure_skip_verify: true
      absolute_time_range:
       start: '2023-01-01T00:00:00Z'
       end:
       truncate: true
``` 
Graph like:
![image](https://github.com/jacksontj/promxy/assets/76882301/10df0ae9-e96a-41b2-afea-bad4d237d4fe)

by looking at Promxy's query_range api, we can see there are only 5 mins data since start time:
![image](https://github.com/jacksontj/promxy/assets/76882301/1e6551be-134d-4781-a8ef-06dfc4dedb8f)

Is there anything that I configured wrongly? 

Thanks for help!

**Comment by jacksontj** [comment]
I imagine this has to do with the query params:

```
        step: 15s
```

As this would add step=15 to *all* downstream queries. The step will already be calculated and passed down -- so there is no need to set this value. The intent of this section is to set some headers (e.g. the nocache one in there).

Otherwise, I have done some testing before with VM and it works fine (and I'm aware of others that use it as well successfully).

Hopefully that does it, if not let me know!

**Comment by yrka5180** [comment]
> I imagine this has to do with the query params:
> 
> ```
>         step: 15s
> ```
> 
> As this would add step=15 to _all_ downstream queries. The step will already be calculated and passed down -- so there is no need to set this value. The intent of this section is to set some headers (e.g. the nocache one in there).
> 
> Otherwise, I have done some testing before with VM and it works fine (and I'm aware of others that use it as well successfully).
> 
> Hopefully that does it, if not let me know!

I removed the query,
```
 step: 15s
```
but it still not work. 

config for now:
```
global:
  scrape_interval: 1m
  scrape_timeout: 10s
  evaluation_interval: 1m
promxy:
  server_groups:
  - remote_read: false
    remote_read_path: api/v1/read
    http_client:
      dial_timeout: 1s
      tls_config:
        insecure_skip_verify: true
      follow_redirects: false
      enable_http2: false
    scheme: http
    labels: {}
    path_prefix: /prometheus
    query_params:
      nocache: "1"
    anti_affinity: 15s
    timeout: 10s
    ignore_error: false
    relative_time_range: null
    absolute_time_range:
      start: 2023-05-23T07:00:00Z
      end: 0001-01-01T00:00:00Z
      truncate: true
    label_filter: null
tls_server_config:
  cert_file: ""
  key_file: ""
  client_auth_type: ""
  client_ca_file: ""
  cipher_suites: []
  curve_preferences: []
  min_version: 0
  max_version: 0
  prefer_server_cipher_suites: false
```

**Comment by yrka5180** [comment]
It seems like If I use **absolute_time_range** and left **end** blank, it will cause this problem.
And if I set **end** to a future time, it works.
![image](https://github.com/jacksontj/promxy/assets/76882301/a339d88b-2025-44ee-affb-0535824277af)

config:
```
promxy:
  server_groups:
  - remote_read: false
    remote_read_path: api/v1/read
    http_client:
      dial_timeout: 1s
      tls_config:
        insecure_skip_verify: true
      follow_redirects: false
      enable_http2: false
    scheme: http
    labels: {}
    path_prefix: /prometheus
    query_params:
      nocache: "1"
      step: 15s
    anti_affinity: 15s
    timeout: 1m0s
    ignore_error: false
    relative_time_range: null
    absolute_time_range:
      start: 2023-05-23T07:00:00Z
      end: 2099-05-23T07:00:00Z
      truncate: true
    label_filter: null
```

**Comment by jacksontj** [comment]
So from your last response here it seems that the root of the issue is how the time filters are applied on the `edge` of the time window. Sure enough there is an issue here -- which I've fixed in #603 . 

I was unable to reproduce the truncated (5m) response you reported -- I always got an error:

```Error executing query: bad_data: invalid parameter "end": end timestamp must not be before start time```


But hopefully that was the issue? If not, then we can debug some more :)

**Comment by yrka5180** [comment]
> So from your last response here it seems that the root of the issue is how the time filters are applied on the `edge` of the time window. Sure enough there is an issue here -- which I've fixed in #603 .
> 
> I was unable to reproduce the truncated (5m) response you reported -- I always got an error:
> 
> `Error executing query: bad_data: invalid parameter "end": end timestamp must not be before start time`
> 
> But hopefully that was the issue? If not, then we can debug some more :)

Yes, that helps. Thanks so much your response!
